MatchServer/GunGame.cpp

```diff
@@ -0,0 +1,132 @@
+#include "stdafx.h"
+#include "GunGame.h"
+
+GunGame::GunGame()
+{
+	m_GGSet.clear();
+}
+
+GunGame::~GunGame()
+{
+	for (auto itor = m_GGSet.begin(); itor != m_GGSet.end(); ++itor)
+	{
+		for (auto vec = (*itor).second.begin(); vec != (*itor).second.end(); ++itor)
+		{
+			delete *vec;
+		}
+	}
+	m_GGSet.clear();
+}
+
+GunGame* GunGame::GetInstance()
+{
+	static GunGame Instance;
+
+	return &Instance;
+}
+
+bool GunGame::ReadXML(const char* szFileName, MZFileSystem* pFileSystem)
+{
+	MXmlDocument	xmlIniData;
+	xmlIniData.Create();
+
+	char *buffer;
+	MZFile mzf;
+
+	if (pFileSystem)
+	{
+		if (!mzf.Open(szFileName, pFileSystem))
+		{
+			if (!mzf.Open(szFileName))
+			{
+				xmlIniData.Destroy();
+				return false;
+			}
+		}
+	}
+	else
+	{
+		if (!mzf.Open(szFileName))
+		{
+			xmlIniData.Destroy();
+			return false;
+		}
+	}
```

Comment:
This whole block is pointless, and could be reduced to `if (!mzf.Open(szFileName)) return false;`.